### PR TITLE
fix nxos launch.py and Dockerfile

### PR DESCRIPTION
**Pull request by lpailhas**
First step to make nxos working on containerlab.

### CODE REVIEW

File: nxos/docker/launch.py [hunk 1]
```diff
@@ -42,6 +42,16 @@ def __init__(self, username, password):
...
                 ['admin', 'admin']
             ]
+        self.hostname = hostname
+        self.conn_mode = conn_mode
+
+    def gen_nics(self):
+        """
+        Override gen_nics by introducing a delay to let eth1+ interfaces to appear
+        """
+        logger.debug("waiting 5sec for container interfaces to settle...")
```
Comment (hellt) [inline nxos/docker/launch.py:52]: Hi @lpailhas 
Thanks a lot for coming up with this PR! It is great to see more platforms coming in.

One thing I wanted to address with other images, that I did for SR OS only so far is eliminating the unconditional wait timer. Apparently, not always interfaces are fast enough to appear inside the docker container and that causes images to fail.

Here is what I did for SR OS https://github.com/hellt/vrnetlab/commit/d29dd4cc3eb09a1e32f6da0beb3a38e0a92588ed

the logic is that I wait for `eth1` to appear, and chances are that the rest of the interfaces will follow rather quick after that.

Can you try this with NX OS?

PS. By waiting for eth1 to appear inside the container we effectively making a requirement to have a data interface present on the node. If you don't want to connect the node to anything, you can simply connect it to a host network stack like that:

```yaml
links:
  - endpoints: ["nxos:eth1", "host:nxos-eth1"]
```

this link definition in containerlab will create a veth link between nxos node `eth1` interface and `nxos-eth1` interface on the host.

Comment (lpailhas) [reply nxos/docker/launch.py]: Hey, to be honest it was I did first, a copy past of what you did for SR OS.
Unfortunately, nxos "kind" doesn't exist on containerlab then you can't run easily the container "alone" to see if everything is ok.
I had already work on containerlab to add nxos but it's not clean :)

I edited gen_nics like for SR OS.

Comment (hellt) [reply nxos/docker/launch.py]: cool, thank you! 
yes, we can help you with containerlab part as well, that should be straightforward

**Comment by hellt** [comment]
@lpailhas which nxos version you tested this with?

**Comment by hellt** [comment]
also, do you know if SSH is working on the version you have?

**Comment by lpailhas** [comment]
Hey, i used titanium-final.7.3.0.D1.1.qcow2, about NETCONF it's not tested.

**Comment by hellt** [comment]
Ok, cool.I have added titanium based nxos to containerlab. it doesn't seem
to have netconf though, or I haven't succeed in enabling it
https://github.com/srl-labs/containerlab/pull/532

But I think what would be needed is supporting a more recent nx9kv nx-os

**Comment by lpailhas** [comment]
Oh really cool, thanks you ! 
 Yes you are right, the best is supporting nx9kv, i am in holiday but i can try this when i'm back.

**Comment by hellt** [comment]
@lpailhas the work for n9kv has been started in #54

**Pull request merged**